pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -276,50 +276,23 @@ def _pin_nvrs(self, nvrs: List[str], releases_config) -> Dict:
         return microshift_entry
 
     async def _create_or_update_pull_request(self, nvrs: List[str]):
-        self._logger.info("Creating ocp-build-data PR...")
-        # Clone ocp-build-data
-        build_data_path = self._working_dir / "ocp-build-data-push"
-        build_data = GitRepository(build_data_path, dry_run=self.runtime.dry_run)
-        ocp_build_data_repo_push_url = self.runtime.config["build_config"]["ocp_build_data_repo_push_url"]
-        await build_data.setup(ocp_build_data_repo_push_url)
         branch = f"auto-pin-microshift-{self.group}-{self.assembly}"
-        await build_data.fetch_switch_branch(branch, self.group)
-        # Make changes
-        releases_yaml_path = build_data_path / "releases.yml"
-        releases_yaml = yaml.load(releases_yaml_path)
-        self._pin_nvrs(nvrs, releases_yaml)
-        yaml.dump(releases_yaml, releases_yaml_path)
-        # Create a PR
         title = f"Pin microshift build for {self.group} {self.assembly}"
         body = f"Created by job run {self.runtime.get_job_run_url()}"
-        match = re.search(r"github\.com[:/](.+)/(.+)(?:.git)?", ocp_build_data_repo_push_url)
-        if not match:
-            raise ValueError(f"Couldn't create a pull request: {ocp_build_data_repo_push_url} is not a valid github repo")
-        head = f"{match[1]}:{branch}"
-        base = self.group
-        if self.runtime.dry_run:
-            self._logger.warning("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'", head, base, title, body)
-            d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
-            result = namedtuple('pull_request', d.keys())(*d.values())
-            return result
-        pushed = await build_data.commit_push(f"{title}\n{body}")
-        result = None
-        if pushed:
-            github_token = os.environ.get('GITHUB_TOKEN')
-            if not github_token:
-                raise ValueError("GITHUB_TOKEN environment variable is required to create a pull request")
-            owner = "openshift-eng"
-            repo = "ocp-build-data"
-            api = GhApi(owner=owner, repo=repo, token=github_token)
-            existing_prs = api.pulls.list(state="open", base=base, head=head)
-            if not existing_prs.items:
-                result = api.pulls.create(head=head, base=base, title=title, body=body, maintainer_can_modify=True)
-            else:
-                pull_number = existing_prs.items[0].number
-                result = api.pulls.update(pull_number=pull_number, title=title, body=body)
-        else:
-            self._logger.warning("PR is not created: Nothing to commit.")
-        return result
+        repo = self.runtime.new_github_client().get_repo("openshift-eng/ocp-build-data")
+        microref = repo.create_git_ref(f"refs/heads/{branch}", repo.get_git_ref(f"heads/{self.group}").object.sha)  # create a microshift branch
```

Comment:
This could fail if the branch already exists.

Reply:
check added